### PR TITLE
Printlist

### PR DESCRIPTION
**Pull request by PotatoParser**
Added more redis publishes containing the following (**everything is serialized before published to redis**):
```json
{
    "user": "username",
    "time": "time of status change",
    "status": "A status code",
    "id": "job id"
}
```
The following are the status codes:
**0** - Print job is completed
**1** - Print job is currently pending
**2** - Print job has been canceled due to a problematic file
**3** - Print job has been canceled due exceeding daily quota of printing
**4** - Print job has encountered an unexpected error

### CODE REVIEW

File: modules/ocf_printhost/files/enforcer [hunk 1]
```diff
@@ -35,6 +37,21 @@ CONF_FILE = '/opt/share/enforcer/enforcer.conf'
 
 Message = namedtuple('Message', ['subject', 'body'])
 
+# Print Job Statuses:
+# 0 - Completed
+# 1 - Pending
+# 2 - Error (problematic file)
+# 3 - Error (insufficient quota)
+# 4 - Error (job processing)
+
```
Comment (ja5087) [inline modules/ocf_printhost/files/enforcer]: nit: extra lines

File: modules/ocf_printhost/files/enforcer [hunk 2]
```diff
@@ -35,6 +37,21 @@ CONF_FILE = '/opt/share/enforcer/enforcer.conf'
 
 Message = namedtuple('Message', ['subject', 'body'])
 
+# Print Job Statuses:
+# 0 - Completed
+# 1 - Pending
+# 2 - Error (problematic file)
+# 3 - Error (insufficient quota)
+# 4 - Error (job processing)
```
Comment (ja5087) [inline modules/ocf_printhost/files/enforcer]: what does this mean?

Comment (PotatoParser) [reply modules/ocf_printhost/files/enforcer]: It’s the numerical representation for status codes

File: modules/ocf_printhost/files/enforcer (hunk 1 above)
Comment (PotatoParser) [reply modules/ocf_printhost/files/enforcer]: Autopep8 fails when removing this newline.

File: modules/ocf_printhost/files/enforcer (hunk 2 above)
Comment (ja5087) [reply modules/ocf_printhost/files/enforcer]: Yes but what does `job processing` mean in relation to `JOB_ERROR`?

File: modules/ocf_printhost/files/enforcer (hunk 1 above)
Comment (ja5087) [reply modules/ocf_printhost/files/enforcer]: I see, it seems PEP8 wants block comments separated by two lines. Maybe either inline the comments next to the line itself? Though I don't mind.

File: modules/ocf_printhost/files/enforcer (hunk 2 above)
Comment (PotatoParser) [reply modules/ocf_printhost/files/enforcer]: Like, an error processing the job

File: modules/ocf_printhost/files/enforcer [hunk 3]
```diff
@@ -167,12 +185,23 @@ def send_printer_mail(message, job, quo):
         semester_pages=quo.semesterly,
         daily_quota=quota.daily_quota(),
         semester_quota=quota.SEMESTERLY_QUOTA,
+        id=job.id
     )
     send_mail_user(job.user, message.subject, body)
 
 
+def statusUpdate(r, job, statusCode):
```
Comment (encadyma) [inline modules/ocf_printhost/files/enforcer]: not really relevant or needed, but maybe `statusUpdate` -> `status_update`/`publish_status` to keep in line with style

**Comment by keur** [comment]
Would definitely recommend you squash these into one commit.

File: modules/ocf_printhost/files/enforcer (hunk 1 above)
Comment (kpengboy) [reply modules/ocf_printhost/files/enforcer]: Consider using inline comments and a doc string?
```python
class PrintJob(Enum):
  """A print job status."""
  COMPLETED = 0
  PENDING = 1
  # Error: problematic file
  FILE_ERROR = 2
  # Error: insufficient quota
  QUOTA_LIMIT = 3
  # Error: job processing
  JOB_ERROR = 4

File: modules/ocf_printhost/files/enforcer [hunk 4]
```diff
@@ -35,6 +37,21 @@ CONF_FILE = '/opt/share/enforcer/enforcer.conf'
...
+# 2 - Error (problematic file)
+# 3 - Error (insufficient quota)
+# 4 - Error (job processing)
+
+
+class PrintJob(Enum):
+    COMPLETED = 0
+    PENDING = 1
+    FILE_ERROR = 2
+    QUOTA_LIMIT = 3
```
Comment (kpengboy) [inline modules/ocf_printhost/files/enforcer]: Nit: `QUOTA_EXCEEDED` might be slightly more descriptive

File: modules/ocf_printhost/files/enforcer [hunk 5]
```diff
@@ -167,12 +185,23 @@ def send_printer_mail(message, job, quo):
         semester_pages=quo.semesterly,
         daily_quota=quota.daily_quota(),
         semester_quota=quota.SEMESTERLY_QUOTA,
+        id=job.id
     )
     send_mail_user(job.user, message.subject, body)
 
 
+def status_update(r, job, statusCode):
```
Comment (kpengboy) [inline modules/ocf_printhost/files/enforcer]: Consider adding a docstring for what this function does.

Comment (kpengboy) [inline modules/ocf_printhost/files/enforcer]: `status_code`

**Pull request merged**